lib/scorecard.rb

```diff
@@ -0,0 +1,95 @@
+
+class ScoreCard
+  attr_reader :frames, :running_score, :frame_score, :scores
+
+  def initialize
+    @frames = 1
+    @running_score = []
+    @frame_score = []
+    @scores = []
+    @cumul_scores = []
+  end
+
+  def run
+    while @frames < 10 do
+      roll_1_input
+      if strike?
+        @frames += 1
+        clear_frame
+      else
+        roll_2_input
+        @frames += 1
```

Comment:
We do
```
@frames += 1
clear_frame
``` 
to the frame in both the if and the else branch, so we could extract this out of the if/else and refactor to look like:
```
if !strike?
    roll_2_input
end
@frames += 1
clear_frame
```
With this format, it's easier for a reader of the code to see that there is logic that happens whether there's a strike or not and DRYs out the code

Reply:
We could do a similar refactor for the 10th roll logic